UnityProject/Assets/Scripts/Objects/Machines/Jukebox.cs

```diff
@@ -168,34 +167,19 @@ private void UpdateMe()
 			// We didn't use "AudioSource.isPlaying" here because of a racing condition between PlayNetworkAtPos latency and Update.
 			if (IsPlaying && Time.time > startPlayTime + musics[currentSongTrackIndex].AudioSource.clip.length)
 			{
-				// The fun isn't over, we just finished the current track.  We just start playing the next one (or stop if it was the last one).
-				if (NextSong() == false)
-				{
-					_ = Stop();
-				}
+				NextSong();
 			}
 		}
 
-		[Server]
-		private async Task WaitForActionToFinish(System.Action action)
-		{
-			actionNotDone = true;
-			await Task.Run(() => action);
-			actionNotDone = false;
-		}
-
 		public async Task Play()
 		{
 			// Too much damage stops the jukebox from being able to play
-			if (integrity.integrity > integrity.initialIntegrity / 2 || actionNotDone)
-			{
-				SoundManager.StopNetworked(guid);
-				IsPlaying = true;
-				spriteHandler.SetSpriteSO(SpritePlaying);
-				guid  = await SoundManager.PlayNetworkedAtPosAsync(musics[currentSongTrackIndex], registerTile.WorldPositionServer, audioSourceParameters, false, true, sourceObj: gameObject);
-				startPlayTime = Time.time;
-				UpdateGUI();
-			}
+			if (integrity.integrity < integrity.initialIntegrity / 2) return;
+			IsPlaying = true;
+			spriteHandler.SetSpriteSO(SpritePlaying);
```

Comment:
this will error from different thread I think

Reply:
Wont error because it's done entirely on the server + hasn't errored before